### DISCUSSION TITLE
Unable to set up with React

### DISCUSSION THREAD

**Discussion by alex-ironside** [faker-js/faker#126] [General]
I installed using `npm i @faker-js/faker -D`

I added the `faker.d.ts` file with this content:

```
// faker.d.ts
declare module "@faker-js/faker" {
  import faker from "faker";
  export default faker;
}
```
But ts doesn't see `faker` as a valid module. What am I missing?

Here is a reproduced sandbox

https://codesandbox.io/s/amazing-smoke-qx63t?file=/src/App.tsx

**Comment by Shinigami92** [comment]
Use `import faker from "@faker-js/faker";`

**Reply by alex-ironside** [reply]
Ok, now the code is not complaining, but no type checking is present. Any ideas? Also, it that's the case, an update here might be in order: [@faker-js/faker](https://github.com/faker-js/faker#nodejs)

**Comment by damienwebdev** [comment]
The Faker we have published today does not have Typescript support "out-of-the-box". There is a maintained `@types/faker` that houses the types. 

We will be publishing (in the next few months) first-party types once we've merged various PRs, but for currently published versions of `@faker-js/faker` to be compatible with Typescript projects you need to:

```bash
npm install -d @faker-js/faker @types/faker
```

And then add

```ts
// faker.d.ts
declare module "@faker-js/faker" {
  import faker from "faker";
  export default faker;
}
```

And then import like

```ts
import faker from "@faker-js/faker";
```

**Reply by alex-ironside** [reply]
After installing both packages, I am getting `Unable to resolve path to module 'faker'.` in "faker" in `import faker from "faker";` in `faker.d.ts`

**Reply by damienwebdev** [reply]
https://codesandbox.io/s/modest-dream-5cyq7

**Reply by damienwebdev** [reply]
It looks like the `animal` type is missing from `@faker-js/faker`

**Comment by damienwebdev** [comment]
@alex3wielki is this still an issue for you?

**Reply by efosaosasu** [reply]
i couldnt use it in my react 
I did evey thing you said

**Reply by ST-DDT** [reply]
Which version of the library and node are you using?
Has the error changed since January?

Note: With current versions, you dont need the extra module/type declarations anymore.
